modules/common-fs2/src/test/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/AssetsSpec.scala

```diff
@@ -116,38 +119,41 @@ class AssetsSpec extends Specification with CatsIO with ScalaCheck {
       val uri = URI.create("http://localhost:8080/maxmind/GeoIP2-City.mmdb")
       val filename = "maxmind"
 
-      Stream
-        .resource(SpecHelpers.refreshState(List(uri -> filename)))
-        .flatMap { state =>
-          val resources =
-            for {
-              blocker <- Blocker[IO]
-              sem <- Resource.eval(Semaphore[IO](1L))
-              enrichments <- Environment.Enrichments.make[IO](List(), BlockerF.noop)
-              _ <- SpecHelpers.filesResource(blocker, TestFiles)
-            } yield (blocker, sem, enrichments)
-
-          val update = Stream
-            .resource(resources)
-            .flatMap {
-              case (blocker, sem, enrichments) =>
-                Assets.updateStream[IO](blocker, sem, state, enrichments, 1.second, List(uri -> filename))
-            }
-            .haltAfter(2.second)
-
-          val before =
-            Stream
-              .eval(state.hashes.get.map(_.get(uri)))
-              .concurrently(update)
-
-          val after = Stream.eval(state.hashes.get.map(_.get(uri)))
-          before ++ update ++ after
-        }
-        .withHttp
-        .haltAfter(3.second)
-        .compile
-        .toList
-        .map(_ == List(Some(Assets.Hash(maxmind1Hash)), (), Some(Assets.Hash(maxmind2Hash))))
+      Clients.mkHttp[IO](ExecutionContext.global).use { httpClient =>
```

Comment:
You might find it interesting to look at [these unit tests in iglu-scala-client](https://github.com/snowplow/iglu-scala-client/blob/e9b8b9c2718a45da0ce4c25d50cea54bdac5dfcf/modules/http4s/src/test/scala/com.snowplowanalytics.iglu/client/resolver/registries/Http4sRegistryLookupSpec.scala#L39-L45).  Remember a `Client[F]` is just a kleisli around a `Request[F] => F[Response[F]]` so you can easily implement something custom in a unit test.  You don't need to instantiate a full Blaze client with an underlying tcp pool.

Here you instantiate a full blaze client

Reply:
That wasn't what I meant 😆.  My point was that in a _unit test_ your client does not need to make real http requests whatsoever.  You can "fake" a client by making it respond with whatever response you want for the test.

```
val client = HttpClient[IO] { _ =>
  Resource.eval(Ok("ok!"))
}
```

It's not a huge problem -- I am only making this suggestion to make you aware of the testing pattern, because I think it is a nice one.

Reply:
ah, my bad, I see the idea of mocking, that's nice, I'll use it :)

Reply:
I don't think of it as mocking.  I think of it as honouring the trait.  In http4s, a client **is** a function from `Request[F]` to `Response[F]`.  And if the function under test takes a client in its signature, then we should be allowed to give it any chosen implementation of `Request[F] => F[Response[F]]`.

To me, mocking means something slightly different.  I think mocking is when you hack at a class to make it pretend to be something that it is not.

Reply:
Replaced all similar uses with thin http clients similar to iglu scala client